src/workerd/tests/bench-jsstring.c++

```diff
@@ -0,0 +1,302 @@
+  TestFixture fixture;
+  fixture.runInIoContext([&](const TestFixture::Environment& env) {
+    auto& js = env.js;
+
+    kj::Vector<char> vec(1024);
+    for (int i = 0; i < 1024; i++) {
+      vec.add('a');
+    }
+    auto str = js.str(vec.asPtr());
+
+    for (auto _: state) {
+      KJ_ASSERT(str.isFlat());
+      benchmark::DoNotOptimize(str.utf8Length(js));
+    }
+  });
+}
+
+// Benchmarks utf8Length on 8192-char latin1 flat strings
+static void JsString_Utf8Length_Latin1_Flat_8192(benchmark::State& state) {
+  TestFixture fixture;
+  fixture.runInIoContext([&](const TestFixture::Environment& env) {
+    auto& js = env.js;
+
+    kj::Vector<char> vec(8192);
+    for (int i = 0; i < 8192; i++) {
+      vec.add('a');
+    }
+    auto str = js.str(vec.asPtr());
+
+    for (auto _: state) {
+      KJ_ASSERT(str.isFlat());
+      benchmark::DoNotOptimize(str.utf8Length(js));
+    }
+  });
+}
+
+// Benchmarks utf8Length on 256-char utf16 flat strings
+static void JsString_Utf8Length_Utf16_Flat_256(benchmark::State& state) {
+  TestFixture fixture;
+  fixture.runInIoContext([&](const TestFixture::Environment& env) {
+    auto& js = env.js;
+
+    kj::Vector<uint16_t> vec(256);
+    for (int i = 0; i < 256; i++) {
+      vec.add(0x1F600 & 0xFFFF);
+    }
+    auto str = js.str(vec.asPtr());
+
+    for (auto _: state) {
+      KJ_ASSERT(str.isFlat());
+      benchmark::DoNotOptimize(str.utf8Length(js));
+    }
+  });
+}
+
+// Benchmarks utf8Length on 1024-char utf16 flat strings
+static void JsString_Utf8Length_Utf16_Flat_1024(benchmark::State& state) {
+  TestFixture fixture;
+  fixture.runInIoContext([&](const TestFixture::Environment& env) {
+    auto& js = env.js;
+
+    kj::Vector<uint16_t> vec(1024);
+    for (int i = 0; i < 1024; i++) {
+      vec.add(0x1F600 & 0xFFFF);
+    }
+    auto str = js.str(vec.asPtr());
+
+    for (auto _: state) {
+      KJ_ASSERT(str.isFlat());
+      benchmark::DoNotOptimize(str.utf8Length(js));
+    }
+  });
+}
+
+// Benchmarks utf8Length on 8192-char utf16 flat strings
+static void JsString_Utf8Length_Utf16_Flat_8192(benchmark::State& state) {
+  TestFixture fixture;
+  fixture.runInIoContext([&](const TestFixture::Environment& env) {
+    auto& js = env.js;
+
+    kj::Vector<uint16_t> vec(8192);
+    for (int i = 0; i < 8192; i++) {
+      vec.add(0x1F600 & 0xFFFF);
+    }
+    auto str = js.str(vec.asPtr());
+
+    for (auto _: state) {
+      KJ_ASSERT(str.isFlat());
+      benchmark::DoNotOptimize(str.utf8Length(js));
+    }
+  });
+}
+
+// Benchmarks utf8Length on 256-char non-flat latin1 strings
+static void JsString_Utf8Length_Latin1_NonFlat_256(benchmark::State& state) {
+  TestFixture fixture;
+  fixture.runInIoContext([&](const TestFixture::Environment& env) {
+    auto& js = env.js;
+
+    kj::Vector<char> vec1(128);
+    kj::Vector<char> vec2(128);
+    for (int i = 0; i < 128; i++) {
+      vec1.add('a');
+      vec2.add('b');
```

Comment:
Nit: If it's latin-1, it would likely be good to have one of the characters be non-ascii